index.js

```diff
@@ -146,8 +146,8 @@ function $asInteger (i) {
 
 function $asNumber (i) {
   var num = Number(i)
-  if (isNaN(num)) {
-    return 'null'
+  if (!isFinite(num)) {
```

Comment:
`!isFinite(null)` will evaluate to `false` and will return `"0"` IMO only strings can be coerced to a number.

Reply:
`isFinite` matches `NaN` and `Infinity`. Both are not used in JSON.

With this implementation, only numeric string and boolean can be coerced to number

Reply:
Yes, but `null` should not be considered as number or fallback to 0. Please look at my updated code. It covers all.